Sming/Components/crypto/include/Crypto/Sha256.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "HashContext.h"
+
+namespace Crypto
+{
+class Sha256Engine
+{
+public:
+	using Hash = HashValue<Sha256Engine>;
+	static constexpr const char* name = "SHA256";
+	static constexpr size_t size = SHA256_SIZE;
+
+	void init()
+	{
+		SHA256_Init(&context);
```

Comment:
`ESP_` prefix for ESP8266 architecture?

Reply:
SHA256 is implemented in libcrypto.a, and benchmarks slower than the axtls version so no need to handle it differently. I'll add a document to summarise.